genesis.py

```diff
@@ -67,75 +83,65 @@ def alert_track(self,chatid,chatname):
-                db.commit()
+                cursor.execute(update, (1, chatid))
                 wrapper.sendMessage(chat_id = chatid, text = "Starting tracking alerts, here we go!\n")
             else:
                 #stops
-
-                update = ("UPDATE users SET track = %s WHERE chat_id = %s ")
-                cursor.execute(update,("0",chatid))
-                db.commit()
+                cursor.execute(update, (0, chatid))
                 wrapper.sendMessage(chat_id = chatid, text = "Stopping the tracking, use /startalerts if you want to start tracking again, operator\n")
+
+            db.commit()
+        db.close()
             
 
-    def invasion_track(self,chatid,chatname):
+    def invasion_track(self,chatid):
         """ Connects to database and change invasion tracking value from 0 to 1 or 1 to 0 (tracking/not tracking) """
 
         db = sqlite3.connect(DB_NAME)
         cursor = db.cursor()
 
-        query = ("SELECT name, invasion_track FROM users WHERE chat_id = %s")
-        cursor.execute(query,(chatid,))
+        query = "SELECT invasion_track FROM users WHERE chat_id = ?"
+        cursor.execute(query, (chatid,))
         row = cursor.fetchone()
 
         if row != None:
             #if fetching was succesful it checks chat_id value
+            update = "UPDATE users SET invasion_track = ? WHERE chat_id = ?"
             
-            if row[1] == "0":
+            if row[0] == 0:
                 #starts
-
-                update = ("UPDATE users SET track = %s WHERE chat_id = %s ")
-                cursor.execute(update,("1",chatid))
-                db.commit()
+                cursor.execute(update, (1, chatid))
                 wrapper.sendMessage(chat_id = chatid, text = "Starting tracking invasions, here we go!\n")
             else:
                 #stops
-
-                update = ("UPDATE users SET track = %s WHERE chat_id = %s ")
-                cursor.execute(update,("0",chatid))
-                db.commit()
+                cursor.execute(update, (0, chatid))
                 wrapper.sendMessage(chat_id = chatid, text = "Stopping the tracking, use /invasion if you want to start tracking again, operator\n")
 
+            db.commit()
+        db.close()
+
     def broadcast(self,field,message):
         """ Broadcast a message checking if a row value is 1 (if users are tracking) """
 
         db = sqlite3.connect(DB_NAME)
         cursor = db.cursor()
 
-        query = ("SELECT chat_id, %s FROM users")
-        cursor.execute(query,(field,))
+        query = "SELECT chat_id FROM users WHERE ? = 1"
+        cursor.execute(query, (field,))
         rows = cursor.fetchall()
 
         for row in rows:
-
-            #check tracking values and send alerts or invasions
-            try:
-                if row[1] == '1':
-                    send_message(row[0],message)
-            except:
-                continue
+            send_message(row[0], message)
```

Comment:
The try/except here is for bypassing html errors sending messages to users that block bot chat without changing track value. It's a fix I have to code because of some users not receiving alerts.